src/stuff.m4

```diff
@@ -444,6 +444,7 @@ dnl $5: Where to get the last value (optional)
 m4_define([_VAL_OPT_ADD_SPACE_WITHOUT_GETOPT_OR_SHORT_OPT], [_JOIN_INDENTED(_INDENT_LEVEL_IN_ARGV_CASE_BODY,
 	[test $[]# -lt 2 && die "Missing value for the optional argument '$_key'." 1],
 	[$3([$1], ["@S|@2"], [$4])],
+	[m4_list_contains([HAVE_SUPPLIED], [$1], [_supplied$4=1], [])],
```

Comment:
You can define a macro `_INDICATION_OF_SUPPLY` in `stuff.m4` for example close to `_APPEND_WRAPPED_ARGUMENT_TO_ARRAY_SPACE`. With body `m4_list_contains([HAVE_SUPPLIED], [$1], [_supplied$2=1], [])` and call it `_INDICATION_OF_SUPPLY([$1], [$4])`